core/services/ocr2/plugins/directrequestocr/aggregation.go

```diff
@@ -0,0 +1,102 @@
+package directrequestocr
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"sort"
+
+	"github.com/smartcontractkit/chainlink/core/services/ocr2/plugins/directrequestocr/config"
+)
+
+func CanAggregate(N int, F int, observations []*ProcessedRequest) bool {
+	return len(observations) >= N-F
+}
+
+func Aggregate(aggMethod config.AggregationMethod, observations []*ProcessedRequest) (*ProcessedRequest, error) {
+	if len(observations) == 0 {
+		return nil, fmt.Errorf("empty observation list passed for aggregation")
+	}
+	var errored []*ProcessedRequest
+	var successful []*ProcessedRequest
+	finalResult := ProcessedRequest{
+		RequestID: observations[0].RequestID,
+		Result:    []byte{},
+		Error:     []byte{},
+	}
+	for _, obs := range observations {
+		if obs.GetError() != nil && len(obs.GetError()) > 0 {
+			errored = append(errored, obs)
+		} else {
+			successful = append(successful, obs)
+		}
+	}
+	var rawData [][]byte
+	if len(errored) > len(successful) {
+		for _, item := range errored {
+			rawData = append(rawData, item.Error)
+		}
+		// Errors are always aggregated using MODE method
+		finalResult.Error = aggregateMode(rawData)
+		return &finalResult, nil
+	}
+	for _, item := range successful {
+		rawData = append(rawData, item.Result)
+	}
+	switch aggMethod {
```

Comment:
A good pattern (strategy?) I did for nodes selection here https://github.com/smartcontractkit/chainlink/pull/7232/files#r949251313
might be a good fit here as well.
Feel free to decline, just an idea.

Reply:
I thought about it but it seemed like an overkill for only two simple functions and no immediate plans to add more. I'll probably leave it for now.